isy-security/src/test/java/de/bund/bva/isyfact/security/oauth2/client/annotation/CustomAuthenticateInterceptorTest.java

```diff
@@ -0,0 +1,53 @@
+package de.bund.bva.isyfact.security.oauth2.client.annotation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.aop.Advisor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ActiveProfiles;
+
+import de.bund.bva.isyfact.security.AbstractOidcProviderTest;
+import de.bund.bva.isyfact.security.autoconfigure.IsyOAuth2ClientAutoConfiguration;
+import de.bund.bva.isyfact.security.oauth2.client.Authentifizierungsmanager;
+
+@ActiveProfiles(profiles = "test-clients")
+@SpringBootTest
+public class CustomAuthenticateInterceptorTest extends AbstractOidcProviderTest {
+
+    @Autowired
+    @Qualifier(IsyOAuth2ClientAutoConfiguration.AUTHENTICATE_INTERCEPTOR)
+    private Advisor authenticateInterceptor;
+
+    private static int customOrder;
+
+    @BeforeAll
+    public static void setup() {
+        registerTestClients();
+    }
+
+    @Test
+    void testAuthenticateInterceptorCanBeChanged() {
+        int order = ((AuthenticateInterceptor) authenticateInterceptor).getOrder();
+        assertEquals(customOrder, order);
+    }
+
+    @TestConfiguration
+    static class TestConfig {
+
+        @Bean(IsyOAuth2ClientAutoConfiguration.AUTHENTICATE_INTERCEPTOR)
+        public Advisor authenticateInterceptor(Authentifizierungsmanager authentifizierungsmanager) {
+            AuthenticateInterceptor custom = new AuthenticateInterceptor(authentifizierungsmanager);
+            customOrder = custom.getOrder() + 500;
+            custom.setOrder(customOrder);
+            return custom;
+        }
+    }
```

Comment:
What's the background of this testcase?
In the class `AuthenticateInterceptor` it is commented:
`Make sure the interceptor runs before Spring annotations like @Secured.`

If the order of the `AuthenticateInterceptor` could be changed afterwards, how is it then ensured, that it will be always processed before the `@Secured`-Annotation?